### PR TITLE
System theme integration for firefox.

### PR DESCRIPTION
**Pull request by 5hubham5ingh**
# Dynamic Color Scheme Based On Browser Theme

## Description
- The color scheme for the new tab will dynamically adapt based on Firefox's UI theme, utilizing its [Theme API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme).
- Rest of the functionality remains the same. 

## Related Issues
- Issue #313

## Results
ezgif com-animated-gif-maker(3)

## Checklist
- [x] I have read and followed the [Contributing Guidelines](../CONTRIBUTING.md).
- [x] I have added the necessary tests.
- [x] The project builds and runs without issues.

### CODE REVIEW

**Comment by itz-rj-here** [comment]
@prem-k-r

**Comment by prem-k-r** [comment]
> # Dynamic Color Scheme Based On Browser Theme
> ## Description
> * The color scheme for the new tab will dynamically adapt based on Firefox's UI theme, utilizing its [Theme API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme).
> * Rest of the functionality remains the same.
> 
> ## Related Issues
> * Issue [Integration with system theme. #313](https://github.com/XengShi/materialYouNewTab/discussions/313)
> 
> ## Results
> ezgif com-animated-gif-maker(3) [ ezgif com-animated-gif-maker(3) ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI) [ ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)

Sorry, I don't use Firefox, so I can't find how you're changing the wallpaper while using the NewTab extension.
Can you guide me, please?

**Comment by XengShi** [comment]
> > # Dynamic Color Scheme Based On Browser Theme
> > ## Description
> > * The color scheme for the new tab will dynamically adapt based on Firefox's UI theme, utilizing its [Theme API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme).
> > * Rest of the functionality remains the same.
> > 
> > ## Related Issues
> > * Issue [Integration with system theme. #313](https://github.com/XengShi/materialYouNewTab/discussions/313)
> > 
> > ## Results
> > ezgif com-animated-gif-maker(3) [ ezgif com-animated-gif-maker(3) ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI) [ ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> 
> Sorry, I don't use Firefox, so I can't find how you're changing the wallpaper while using the NewTab extension.
> Can you guide me, please?
> 
> 

You can run two or more than two apps at the same time in your PC

**Comment by prem-k-r** [comment]
Desktop's wallpaper?

**Comment by XengShi** [comment]
> Desktop's wallpaper? 

Yes, he is probably changing desktop wallpaper

**Comment by 5hubham5ingh** [comment]
I'm using pywalfox extension for the browser's dynamic theme change.

**Comment by 5hubham5ingh** [comment]
> > Desktop's wallpaper? 
> 
> Yes, he is probably changing desktop wallpaper
> 
> Or maybe he opened another window of firefox!!!

I'm using [this](https://github.com/5hubham5ingh/WallRizz) with pywalfox extension that changes the browser theme based on the wallpaper which triggers the theme change event handler in script.js to update the theme using updated browser colours.

**Comment by 5hubham5ingh** [comment]
> > # Dynamic Color Scheme Based On Browser Theme
> > ## Description
> > 
> > * The color scheme for the new tab will dynamically adapt based on Firefox's UI theme, utilizing its [Theme API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme).
> > * Rest of the functionality remains the same.
> > 
> > ## Related Issues
> > 
> > * Issue [Integration with system theme. #313](https://github.com/XengShi/materialYouNewTab/discussions/313)
> > 
> > ## Results
> > ezgif com-animated-gif-maker(3)
> >       
> >         [
> >           
> >             ezgif com-animated-gif-maker(3)
> >           
> >         ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> >         
> >        [ ezgif com-animated-gif-maker(3) ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> >       
> >         [
> >           
> >             ezgif com-animated-gif-maker(3)
> >           
> >         ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> >         
> >        [ ](https://private-user-images.githubusercontent.com/128460722/395819885-e4457452-4366-4e8a-8696-a88f5a39035c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNTU5NjksIm5iZiI6MTczNDI1NTY2OSwicGF0aCI6Ii8xMjg0NjA3MjIvMzk1ODE5ODg1LWU0NDU3NDUyLTQzNjYtNGU4YS04Njk2LWE4OGY1YTM5MDM1Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNVQwOTQxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzgxMmY4NzliZThjYmU3ODhmOTNhMjA1NTA5OWMwNDc4ZDEwZTZhNzk0NzMxODdjMDUzYjE3Yjk3ZGYzZGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.je9pLq59LxW2dsVz2bt6ZmfZfgu9zn5RSM1GtX1UUgI)
> 
> Sorry, I don't use Firefox, so I can't find how you're changing the wallpaper while using the NewTab extension. Can you guide me, please?

Were you able to successfully reproduce and test the setup?

**Comment by XengShi** [comment]
https://github.com/user-attachments/assets/4c0de7da-5ab1-4237-952b-ef758c3b40a5

Changing Firefox theme is affecting is very badly

**Comment by 5hubham5ingh** [comment]
https://github.com/user-attachments/assets/283c6497-a225-430a-8210-6e3cca5964dc

**Comment by prem-k-r** [comment]
> > Sorry, I don't use Firefox, so I can't find how you're changing the wallpaper while using the NewTab extension. Can you guide me, please?
> 
> Were you able to successfully reproduce and test the setup?

Actually, no. I installed the Pywalfox extension from the store, but there was some issue. I wasn’t able to do anything.
Maybe I needed to install the GitHub version?

**Comment by prem-k-r** [comment]
Um . Maybe merge this in firefox branch?

**Comment by 5hubham5ingh** [comment]
> Maybe I needed to install the GitHub version?

Depends on the issue, if the problem lies in configuring it correctly to work in your system then the GitHub version won't fix it.
I suggest you read its docs and see where the problem is.

**Comment by 5hubham5ingh** [comment]
> Um . Maybe merge this in firefox branch? 

Sure, should I send pr for that branch?

**Comment by prem-k-r** [comment]
Let's wait for XengShi

**Comment by XengShi** [comment]
> > Um . Maybe merge this in firefox branch?
> 
> Sure, should I send pr for that branch?

Screenshot From 2024-12-18 13-49-56

For certain themes, it can still result in a poor user experience due to detecting unusual color schemes.

I would recommend adding a toggle button, like this:

    Title: 'Enable browser-based theming'
    Description: 'Extension may  pick messy or unappealing colors.'

**Comment by Thunder-Blaze** [comment]
I would like to add that toggle if you are not working on it @5hubham5ingh
else if you are working on it, Continue

**Comment by 5hubham5ingh** [comment]
> I would like to add that toggle if you are not working on it @5hubham5ingh, I'll make a PR in your forked repo (just sync it first so that there won't be any merge conflicts)
> 
> else if you are working on it, Continue

I have synced the fork, you can implement the toggle feature.

**Comment by prem-k-r** [comment]
https://github.com/XengShi/materialYouNewTab/tree/firefox
this branch goes to Firefox add-ons store
Since we are adding toggle too.. maybe merge in this branch?

**Comment by Thunder-Blaze** [comment]
@5hubham5ingh I had one more thing to ask of you, could you create a copy of the main branch named Feature/ThemeIntegration as I am somehow unable to fork your repo and when I am trying to edit it, it is getting merged into main branch of my own fork

Sorry about that, I am new to this 😅

**Comment by 5hubham5ingh** [comment]
> https://github.com/XengShi/materialYouNewTab/tree/firefox this branch goes to Firefox add-ons store Since we are adding toggle too.. maybe merge in this branch?

Okay, I will send the pr on Firefox branch.

**Comment by Thunder-Blaze** [comment]
hmm, is this line linux specific, cuz its giving me error

image

image

I'll switch to my Linux Workspace if it is, anyway toggle added, just have to fix its functioning

image

I've kept it disabled on Non Firefox Browsers

**Comment by 5hubham5ingh** [comment]
> hmm, is this line linux specific, cuz its giving me error

No, it is not. Make sure to load the extension as extension, using about:debugging, instead of just a web page because browser object is part of web extension api, so it's not available to web pages.

**Comment by Thunder-Blaze** [comment]
We ended up with merge conflicts lol ☠️

**Comment by 5hubham5ingh** [comment]
I have resolved the conflict and tested the extension. @Thunder-Blaze could you also test and see if there's some issue or not?

**Comment by prem-k-r** [comment]
Please check I think this got reverted during fixing conflicts 
https://github.com/XengShi/materialYouNewTab/pull/318/commits/095b42cddd3aef42a4d3595099a5e07e200b9199

File: index.html [hunk 1]
```diff
@@ -950,6 +965,17 @@ <h1>Material You NewTab</h1>
                             </label>
                         </div>
 
+                        <div class="ttcont" id="firefoxAdaptiveField">
+                            <div class="texts">
+                                <div class="bigText" id="firefoxAdaptiveText">Enable Firefox Theme</div>
+                                <div class="infoText" id="firefoxAdaptiveInfoText">Experimental</div>
```
Comment (prem-k-r) [inline index.html:971]: It will be better if it is more user friendly and descriptive

Comment (Thunder-Blaze) [reply index.html:971]: Hmm, I tried writing what @XengShi exactly said but then the text was intermixing with the text of top and bottom checkbox so had to decrease the words

File: script.js [hunk 2]
```diff
@@ -1503,8 +1731,8 @@ const handleColorPickerChange = function (event) {
 };
 
 // Add listeners for color picker
-colorPicker.removeEventListener('change', handleColorPickerChange); // Ensure no duplicate listeners
-colorPicker.addEventListener('change', handleColorPickerChange);
+colorPicker.removeEventListener('input', handleColorPickerChange); // Ensure no duplicate listeners
+colorPicker.addEventListener('input', handleColorPickerChange);
```
Comment (Thunder-Blaze) [inline script.js:1735]: Yeah this have gotten reverted

File: index.html (hunk 1 above)
Comment (Thunder-Blaze) [reply index.html:971]: Title: 'Enable Browser Theming'
Description: 'Extension may pick messy colors.'

how about this ? this won't overflow

**Comment by Thunder-Blaze** [comment]
There is also some error in weather related code
image

File: index.html (hunk 1 above)
Comment (prem-k-r) [reply index.html:971]: Enable Browser-based Theming

Theme color will be adapted from the browser

**Comment by prem-k-r** [comment]
> There is also some error in weather related code image
> 
> Though this happens only on the first occurence idk why
> 
> Menu button color in dark mode is also wrong image

Some other changes are also not reflecting in the code..

**Comment by Thunder-Blaze** [comment]
Sure, but I'll need some time maybe till tomorrow as I am working on my own repo [BlazinLock](https://github.com/Thunder-Blaze/BlazinLock) rn

**Comment by prem-k-r** [comment]
> > Some other changes are also not reflecting in the code..
> 
> Could you please make a PR for those? Also @Thunder-Blaze

ok

File: index.html (hunk 1 above)
Comment (Thunder-Blaze) [reply index.html:971]: @prem-k-r this is what I was saying, the text is going out of bounds, should I keep it like this or can you suggest a smaller title ig?

image

Comment (prem-k-r) [reply index.html:971]: Try
Colors will be adapted from browser

Comment (Thunder-Blaze) [reply index.html:971]: image
hmm better, can we make the title a bit smaller too ?

Comment (prem-k-r) [reply index.html:971]: Remove "Enable"

**Comment by Thunder-Blaze** [comment]
@5hubham5ingh try creating pr from Feature/ThemeIntegration branch after merging my PR, I don't think we should have conflicts anymore, but we can never be sure

btw I put PR in Feature/ThemeIntegration as in main branch I was having merge conflicts

**Comment by Thunder-Blaze** [comment]
If we still have issues, then
1. Create a new folder and pull the firefox branch
2. Copy paste all the files from my Feature/ThemeIntegration branch into that folder
3. Commit the changes and push them to your own repo under firefox branch
4. And them make the final PR

I'm sure this won't produce a single merge conflict but it is the last resort kind of ig

File: index.html (hunk 1 above)
Comment (Thunder-Blaze) [reply index.html:971]: done, looks good now

**Comment by Thunder-Blaze** [comment]
@5hubham5ingh just merge the PR I sent on the main branch, that should so finally

**Comment by 5hubham5ingh** [comment]
> @5hubham5ingh just merge the PR I sent on the main branch, that should do everything finally
Merged and tested, working fine on my system.

**Comment by Thunder-Blaze** [comment]
@prem-k-r can you confirm the changes?
Sorry about the big diffs, take your time

**Review by Thunder-Blaze — APPROVED** [review]

**Comment by 5hubham5ingh** [comment]
@XengShi   Since the PR has received approval. Is there anything else needed before it can be merged?

**Pull request merged**